code.js

```diff
@@ -11,53 +15,137 @@ var points = [
 var voyager = { x: 150, y: 300 };
 var thread = null;
 
-function plot(x, y, size, color) {
+var isRunning = false;
+
+var pointsAreUserDefined = false;
+
+function plot(canvas, x, y, size, color) {
+    if(!color) color = "#FF0000";
+    if(!size) size = 1;
+
+    canvas.beginPath();
+    canvas.fillStyle = color;
+    canvas.fillRect(x - size / 2 , y - size / 2, size, size);
+    canvas.fill();
+}
+
+function plotDrawing(x, y, size, color) {
+    plot(ctxDrawing, x, y, size, color);
+}
+
+function plotUI(x, y, size, color) {
+    plot(ctxUI, x, y, size, color);
+}
+
+function drawCircle(ctx, x, y, thickness, size, color) {
     if(!color) color = "#FF0000";
     if(!size) size = 1;
 
-    ctx.fillStyle = color;
-    ctx.fillRect(x,y,size,size);
-    ctx.fill();
+    ctx.beginPath();
```

Comment:
This draws a circle with a line, but doesn't fill inside. Is that the behavior we want?